src/arithmetic.jl

```diff
@@ -40,17 +32,11 @@ end
 
 ## zero and one ##
 for T in (:Taylor1, :TaylorN), f in (:zero, :one)
-    @eval begin
-        ($f)(a::$T) = $T([($f)(a[1])], a.order)
-
-        ($f)(a::$T, order) = $T([($f)(a[1])], order)
-    end
+    @eval ($f)(a::$T) = $T(($f)(a[1]), a.order)
```

Comment:
I fully agree. I have done some work along this, but it is largely unfinished. Actually, after reading different sources, it seems to me that it is better (in terms of memory) to have `Taylor1{Taylor1{T}}`, which effectively is a `TaylorN{T}` of two variables.